catalog/rocks/pgiterator.go

```diff
@@ -0,0 +1,288 @@
+package rocks
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/db"
+)
+
+const (
+	// IteratorPrefetchSize is the amount of records to prefetch from PG
+	IteratorPrefetchSize = 1000
+)
+
+type pgBranchRecord struct {
+	BranchID `db:"id"`
+	*PgBranch
+}
+
+type PGRepositoryIterator struct {
+	db  db.Database
+	ctx context.Context
+
+	value *RepositoryRecord
+	buf   []*RepositoryRecord
+
+	offset           string
+	prefetchSize     int
+	prefetchRequired bool
```

Comment:
This field has a confusing name.  AFAICT it behaves as though it could be called `dontNeedToSelect`; if so, could you replace it with a field with a name more like `needToSelect`?

Reply:
not sure I understand. it indicates wheteher a call to prefetch is required to advance the cursor. to me that makes sense (but again, not opinionated on that, so I'm ok with whatever concrete change you suggest)